client/keys.go

```diff
@@ -91,6 +90,7 @@ func (key *PrivateKey) execute(op gokeyless.Op, msg []byte) ([]byte, error) {
 		SNI:      key.sni,
 	})
 	if err != nil {
+		defer conn.Close()
```

Comment:
Does this need to be a `defer` here since you are returning anyway? Is the conn being kept open when no error has occurred?